src/create-probot.ts

```diff
@@ -38,15 +40,16 @@ export function createProbot({
   defaults = {},
   env = process.env,
 }: CreateProbotOptions = {}) {
-  const privateKey = getPrivateKey({ env });
-  const envWithDefaults = { ...DEFAULTS, ...env };
+  const privateKey = getPrivateKey({ env: { ...process.env, ...env } });
+  const envWithDefaults = { ...DEFAULTS, ...process.env, ...env };
```

Comment:
we shouldn't mix `process.env` and `env` here. We default the `env` argument to `process.env`